pkg/controller/workload/workload_processor.go

```diff
@@ -567,32 +574,22 @@ func (p *Processor) updateEndpointOneByOne(serviceId uint32, epsUpdate []cache.E
 	return nil
 }
 
+// updateEndpoint is called when service lb policy is changed to update the endpoint priority.
 // toLLb indicates whether we are performing a locality load balance update.
 // If toLLb is true, it means we need to calculate priority; otherwise,
 // it represents a random strategy, in which case we just set the priority to 0.
 func (p *Processor) updateEndpoint(serviceId uint32, toLLb bool) error {
 	endpoints := p.EndpointCache.List(serviceId)
+	endpointSlice := make([]cache.Endpoint, 0, len(endpoints))
+	for _, endpoint := range endpoints {
+		endpointSlice = append(endpointSlice, endpoint)
+	}
 	if toLLb {
-		endpointSlice := []cache.Endpoint{}
-		for _, endpoint := range endpoints {
-			endpointSlice = append(endpointSlice, endpoint)
-		}
-
-		// sort by Prio and BackendIndex
-		sort.Slice(endpointSlice, func(i, j int) bool {
-			if endpointSlice[i].Prio == endpointSlice[j].Prio {
-				return endpointSlice[i].BackendIndex < endpointSlice[j].BackendIndex
-			}
-			return endpointSlice[i].Prio < endpointSlice[j].Prio
-		})
 		return p.updateEndpointOneByOne(serviceId, endpointSlice, toLLb)
 	} else {
-		filtered := []cache.Endpoint{}
-		for _, endpoint := range endpoints {
-			if endpoint.Prio > 0 {
-				filtered = append(filtered, endpoint)
-			}
-		}
+		filtered := slices.Filter(endpointSlice, func(e cache.Endpoint) bool {
+			return e.Prio > 0
```

Comment:
Can this return value be used as a function error return?

Reply:
what do you mean

Reply:
My mistake. This one is right.